streams-contrib/streams-persist-hdfs/src/main/java/org/apache/streams/hdfs/WebHdfsPersistReader.java

```diff
@@ -204,6 +212,83 @@ public StreamsResultSet readCurrent() {
         return current;
     }
 
+    public StreamsDatum processLine(String line) {
+
+        String[] fields = line.split(hdfsConfiguration.getFieldDelimiter());
+
+        if( fields.length == 0)
+            return null;
+
+        String id = null;
+        DateTime ts = null;
+        Map<String, Object> metadata = null;
+        String json = null;
+
+        if( hdfsConfiguration.getFields().contains( HdfsConstants.DOC )
```

Comment:
this would be a lot more readable if you pulled out `hdfsConfiguration.getFields()` into its own variable